addons/google_account/models/google_service.py

```diff
@@ -120,6 +120,41 @@ def _get_google_token_json(self, authorize_code, service):
             error_msg = _("Something went wrong during your token generation. Maybe your Authorization Code is invalid")
             raise self.env['res.config.settings'].get_config_warning(error_msg)
 
+    @api.model
+    def _get_access_token(self, refresh_token, service, scope):
+        """Fetch the access token thanks to the refresh token."""
+        get_config_warning = self.env['res.config.settings'].get_config_warning
+        get_param = self.env['ir.config_parameter'].sudo().get_param
+        client_id = get_param('google_%s_client_id' % service, default=False)
+        client_secret = get_param('google_%s_client_secret' % service, default=False)
+
+        if not client_id or not client_secret:
+            raise get_config_warning(_('Google %s is not yet configured.', service.title()))
+
+        if not refresh_token:
+            raise get_config_warning(_('The refresh token for authentication is not set.'))
+
+        try:
+            result = requests.post(
```

Comment:
Check if it's actually necessary to generate a token every time.
For social_youtube, we receive a "expires_in" parameter in the response and can use the same token for a limited amount of time before having to regenerate one.
(For reference: `def _refresh_youtube_token`)

Could save some requests and improve performances (at the expense of a few more fields stored).